BraveRewards/Ledger/BATBraveLedger.mm

```diff
@@ -757,7 +834,205 @@ - (void)confirmationsTransactionHistoryDidChange
+    }
+  }
+  */
+}
+
+- (void)showAddFundsNotificationIfNeccessary
+{
+  const auto stamp = ledger->GetReconcileStamp();
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  // Show add funds notification if reconciliation will occur in the
+  // next 3 days and balance is too low.
+  if (stamp - now > 3 * 24 * 60) {
+    return;
+  }
+  // Make sure it hasnt already been shown
+  const auto upcomingAddFundsNotificationTime = [self.prefs[kNextAddFundsNotificationKey] doubleValue];
+  if (upcomingAddFundsNotificationTime != 0.0 &&
+      now < upcomingAddFundsNotificationTime) {
+    return;
+  }
+  // Make sure they don't have a sufficient balance
+  if (self.hasSufficientBalanceToReconcile) {
+    return;
+  }
+  
+  // Set next add funds notification in 3 days
+  const auto nextTime = [[NSDate date] timeIntervalSince1970] + (60 * 60 * 24 * 3);
+  self.prefs[kNextAddFundsNotificationKey] = @(nextTime);
+  [self savePrefs];
+  
+  [self addNotificationOfKind:BATRewardsNotificationKindInsufficientFunds
+                    arguments:nil
+               notificationID:@"rewards_notification_insufficient_funds"];
+}
+
+- (void)showTipsProcessedNotificationIfNeccessary
+{
+  if (!self.autoContributeEnabled) {
+    return;
+  }
+  [self addNotificationOfKind:BATRewardsNotificationKindTipsProcessed
+                    arguments:nil
+               notificationID:@"rewards_notification_tips_processed"];
+}
+
+- (void)addNotificationOfKind:(BATRewardsNotificationKind)kind
+                    arguments:(nullable NSArray<NSString *> *)arguments
+               notificationID:(nullable NSString *)identifier
+{
+  [self addNotificationOfKind:kind arguments:arguments notificationID:identifier onlyOnce:NO];
+}
+
+- (void)addNotificationOfKind:(BATRewardsNotificationKind)kind
+                    arguments:(nullable NSArray<NSString *> *)arguments
+               notificationID:(nullable NSString *)identifier
+                     onlyOnce:(BOOL)onlyOnce
+{
+  NSParameterAssert(kind != BATRewardsNotificationKindInvalid);
+  NSString *notificationID = [identifier copy];
+  if (!identifier || identifier.length == 0) {
+    notificationID = [NSUUID UUID].UUIDString;
+  } else if (onlyOnce) {
+    const auto idx = [self.mNotifications indexOfObjectPassingTest:^BOOL(BATRewardsNotification * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+      return obj.displayed && [obj.id isEqualToString:identifier];
+    }];
+    if (idx != NSNotFound) {
+      return;
+    }
+  }
+  
+  const auto notification = [[BATRewardsNotification alloc] initWithID:notificationID
+                                                             dateAdded:[[NSDate date] timeIntervalSince1970]
+                                                                  kind:kind
+                                                             arguments:arguments];
+  if (onlyOnce) {
+    notification.displayed = YES;
+  }
+  
+  [self.mNotifications addObject:notification];
```

Comment:
These notifications never launch, only `BATBraveLedgerNotificationAdded` is posted, are you going to add it in another PR or here? Maybe i'm missing something

Reply:
These notifications won't show up as regular apple notifications, they'll update the badge (#66),  and show new UI on the panel when the user next shows it (#45)